test/checkImageSize.js

```diff
@@ -0,0 +1,118 @@
+const sizeOf = require('image-size');
+const fs = require('fs');
+const { exit } = require("process");
+
+function getFileSizeMB(fileStats){
+    return (fileStats.size) / (1024*1024);
+}
+
+//Check if image is larger than 1MB
+function isFileSizeBig(fileStats){ 
+    return getFileSizeMB(fileStats) > 1;
+}
+
+function isImageWidthLarge(imageSize){
+    let width = imageSize.width;
+    return (width >1024);
+}
+
+function isImageWidthSmall(imageSize){
+    let width = imageSize.width;
+    return (width <512);
+}
+
+function isAValidExtension(imageFileName){
+    return (imageFileName.includes(".jpg") | imageFileName.includes(".png") | imageFileName.includes(".jpeg"));
+}
+
+const errorType =  {
+    ASPECT_RATIO_ERROR: {
+        error: true,
+        message: "Not a square aspect ratio",
+        fixable: false
+    },
+    SMALL_IMAGE_ERROR: {
+        error: true,
+        message: "Your image is too small and should be 512px by 512px",
+        fixable: false
+    },
+    LARGE_IMAGE_ERROR: {
+        error: true,
+        message: "Your image is too large and should be 512px by 512px",
+        fixable: true
+    },
+    EXTENSION_ERROR: {
+        error: true,
+        message: "Image File Name does not have lower case extension",
+        fixable: true
+    },
+    FILE_SIZE_ERROR: {
+        error: true,
+        message: "Your file should be below 1MB",
+        fixable: true
+    },
+    NO_ERROR: {
+        error: false,
+        message: "No Error"
+    }
+}
+
+function imageSizeChecker(input){  //input here is the image file
+    //let inputName = "./img/" + input;
+    let imageSize = sizeOf(input);
+    let width = imageSize.width;
+    let height = imageSize.height;
+    let imageStats = fs.statSync(input);
+    if (width != height){
+        return errorType.ASPECT_RATIO_ERROR;
+    }
+    else if (isImageWidthSmall(imageSize)){
+        return errorType.SMALL_IMAGE_ERROR;
+    }
+    else if (isFileSizeBig(imageStats)){
+        return errorType.FILE_SIZE_ERROR;
+    } 
+    else if(isImageWidthLarge(imageSize)){
+        return errorType.LARGE_IMAGE_ERROR;
+    }
+    else if (!isAValidExtension(input)){
```

Comment:
This if statement would also fail if the file had a non .jpg, .jpeg, or .png extension (ie. it would fail if they had a .svg file) However, the error message for EXTENSION_ERROR only mentions file name not having a lower case extension name.

Reply:
The getAllFiles() function(line 86) deals with this since it returns an array of the file names that have .jpg, jpeg, or .png. Therefore, the array won't contain any file that has a non .jpg, .jpeg, or .png extension. We will run imageSizeChecker to files that only have the .jpg, .jpeg, or .png extension.